worlds/__init__.py

```diff
@@ -95,27 +100,94 @@ def load(self) -> bool:
-    "games": {world_name: world.get_data_package_data() for world_name, world in AutoWorldRegister.world_types.items()},
-}
+def scan_worlds() -> List[WorldSource]:
+    # find potential world containers, currently folders and zip-importable .apworld's
+    sources: List[WorldSource] = []
+    for folder in (folder for folder in (user_folder, local_folder) if folder):
+        relative = folder == local_folder
+        for entry in os.scandir(folder):
+            # prevent loading of __pycache__ and allow _* for non-world folders, disable files/folders starting with "."
+            if not entry.name.startswith(("_", ".")):
+                file_name = entry.name if relative else os.path.join(folder, entry.name)
+                if entry.is_dir():
+                    sources.append(WorldSource(file_name, relative=relative))
+                elif entry.is_file() and entry.name.endswith(".apworld"):
+                    sources.append(WorldSource(file_name, is_zip=True, relative=relative))
+    sources.sort()
+    return sources
+
+
+def get_world_by_name(name: str) -> str:
+    for world in world_data:
+        if world["game"] == name:
+            return world
+    else:
+        raise ModuleNotFoundError(f"No game found that matches {name}")
+
+
+def get_worlds_info() -> Tuple[Dict[str, Dict[str, str]], bool]:
+    should_update = False
+    world_data = {}
+    try:
+        cached_time = os.path.getmtime(cached_worlds_path)
+        world_data = orjson.loads(pkgutil.get_data(__name__, cached_worlds_path))
+        for world in world_sources:
+            if os.path.getmtime(world.resolved_path) > cached_time:
+                should_update = True
+                break
+    except (FileNotFoundError, orjson.JSONDecodeError):
+        should_update = True
+
+    return world_data, should_update
+
+
+def load_all_worlds() -> DataPackage:
+    # import all submodules to trigger AutoWorldRegister
+    games: Dict[str, GamesPackage] = {}
+    json_data: Dict[str, Dict[str, str]] = {}
+    for world_source in world_sources:
+        world_source.load()
+
+    for world_name, world in AutoWorldRegister.world_types.items():
+        games[world_name] = world.get_data_package_data()
+        # GitHub runners won't allow me to write to their cache
+        if "runner" in cached_worlds_path:
+            continue
+        json_name = [name for name in world.__module__.split(".") if "world" not in name][0]
+        if world.zip_path:
+            json_name += ".apworld"
+        world_json = {"game": world.game, "path": world.__file__,
+                      "settings": str(world.settings_key) if hasattr(world, "settings") else None}
+        json_data[json_name] = world_json
+
+    if "runner" not in cached_worlds_path:
+        with open(cached_worlds_path, "wb") as f:
+            f.write(orjson.dumps(json_data))
+    return DataPackage(games=games)
+
+
+def load_worlds(games: Union[Iterable[str], str]) -> DataPackage:
```

Comment:
```suggestion
def load_worlds(games: Union[Container[str], str]) -> DataPackage:
```
We don't iterate through `games`. We check to see if `games` contains something.
(needs the import change too)